mautic/mautic
# Fix/saml integration

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [x]
| New feature/enhancement? (use the a.x branch)      | [ ]
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | [ ] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #13361 



#### Description:

The SAML integration is broken, due to an incompatibility with the expected arguments in the `LightSaml\SpBundle` `DefaultController`.
This is addressed in https://github.com/mautic/SpBundle/pull/2, and this PR is the proof that this works.

~~**This PR should not be merged, but is a way to test this out.**~~
**this PR had to be merged, as there is a needed change to address an issue with error handling for SAML errors.**



#### Steps to test this PR:

This PR should be tested **locally** as it requires an usable IDP to test the SAML auth flow.


1. checkout this branch and run `composer update --lock` to get the changed lightsaml/sp-bundle dependency.
2. Run a Mock simplesamlphp instance to be able to test the flow
    Replace `<LOCAL_MAUTIC_URL>` with your local url, e.g. `http://mautic.localhost`
    change the ports if needed if they conflict with your local setup. 
    
    ```
    docker run -p 8080:8080 \
    -e SIMPLESAMLPHP_SP_ENTITY_ID=<LOCAL_MAUTIC_URL> \
    -e SIMPLESAMLPHP_SP_ASSERTION_CONSUMER_SERVICE=<LOCAL_MAUTIC_URL>/s/saml/login_check \
    -d kenchan0130/simplesamlphp
    ```
3. go to `http://localhost:8080/simplesaml/saml2/idp/metadata.php?output=xhtml` and copy the metadata in xml format to a temp file (e.g. `mautic_saml_test_metatada.xml`).
4. go to `<LOCAL_MAUTIC_URL>/s/config/edit?tab=userconfig` and select your entity ID, upload the metadata.xml file, select a role to create users,  and fill in `email` in the fields `Email`, `First Name` and `Last name` (we are testing, so no problem that the actual values are not correct). Click `save`.
5. go in an anonymous tab to `<LOCAL_MAUTIC_URL>` and you should be redirected to the simplesaml page (if you get there, that proofs this patch works)
6. login with username `user1` and password `password` 
7. you're redirected to Mautic and logged in as user1
